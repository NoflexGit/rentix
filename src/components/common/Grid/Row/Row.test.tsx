import React from 'react';
import { render } from '@testing-library/react';
import { Col, Row } from '../index';

const testId = 'test-id';

describe('Row', () => {
  describe('Snapshots tests', () => {
    it('should match snapshot', () => {
      const { container } = render(
        <Row>
          <Col>Example</Col>
        </Row>,
      );

      expect(container).toMatchSnapshot();
    });
  });

  describe('Classes tests', () => {
    it('should set `className` class', () => {
      const className = 'test-className';

      const { container } = render(<Row className={className} />);

      expect(container.firstElementChild).toHaveClass(className);
    });
  });

  it('should set `data-test-id` attribute to col', () => {
    const { getByTestId } = render(<Row dataTestId={testId} />);

    expect(getByTestId(testId).tagName).toBe('DIV');
  });

  it('should set a correct tag', () => {
    const { getByTestId } = render(<Row tag="article" dataTestId={testId} />);

    expect(getByTestId(testId).tagName).toBe('ARTICLE');
  });
});
