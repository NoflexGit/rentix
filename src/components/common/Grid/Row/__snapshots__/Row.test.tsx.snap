// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row Snapshots tests should match snapshot 1`] = `
<div>
  <div
    class="row"
  >
    <div
      class="component width-undefined"
    >
      Example
    </div>
  </div>
</div>
`;
